File: tests/test_pool.js
const fs = require('fs');
const code = fs.readFileSync('src/js/data.js', 'utf8');
const matchW = code.match(/const WEAPONS=\[(.*?)\];/s);
const WEAPONS = eval('[' + matchW[1] + ']');
const matchA = code.match(/const ARMORS=\[(.*?)\];/s);
const ARMORS = eval('[' + matchA[1] + ']');
const matchP = code.match(/const POTIONS=\[(.*?)\];/s);
const POTIONS = eval('[' + matchP[1] + ']');

function isClassUsableGear(item, playerClass){
  if(!item || (item.type !== 'weapon' && item.type !== 'armor')) return false;
  if(item.reqClass && !item.reqClass.includes(playerClass)) return false;
  return true;
}

let lvl = 1;
let playerClass = 'monk';

function gearPoolForPlayer(items, playerClass, levelSlack = 2){
  return items.filter(item =>
    (!item.reqLvl || lvl >= item.reqLvl - levelSlack) &&
    isClassUsableGear(item, playerClass)
  );
}

let weaponPool = gearPoolForPlayer(WEAPONS, playerClass);
let armorPool = gearPoolForPlayer(ARMORS, playerClass);

let pool = [];
pool.push(...weaponPool.map(w=>w.name));
pool.push(...armorPool.map(a=>a.name));
pool.push(...POTIONS.map(p=>p.name));
pool.push(...POTIONS.filter(p=>p.type==='potion').map(p=>p.name));

console.log("JS POOL FOR MONK:");
console.log(pool);
console.log("Length:", pool.length);

File: tests/test_pool.py
import json
from automation.nn_rl.game_engine import WEAPONS, ARMORS, POTIONS

def _is_class_usable_gear(item, player_class):
    if not item or item.get('type') not in ('weapon', 'armor'):
        return False
    req = item.get('reqClass')
    if req and player_class not in req:
        return False
    return True

lvl = 1
player_class = 'monk'

weapon_pool = [w['name'] for w in WEAPONS if _is_class_usable_gear(w, player_class) and (not w.get('reqLvl') or lvl >= w['reqLvl'] - 2)]
armor_pool = [a['name'] for a in ARMORS if _is_class_usable_gear(a, player_class) and (not a.get('reqLvl') or lvl >= a['reqLvl'] - 2)]
potion_names = [p['name'] for p in POTIONS]
healing_names = [p['name'] for p in POTIONS if p['type'] == 'potion']

pool = []
pool.extend(weapon_pool)
pool.extend(armor_pool)
pool.extend(potion_names)
pool.extend(healing_names)

print("PYTHON POOL FOR MONK (forceHigh=True or rng.ch(0.3)):")
print(pool)
print("Length:", len(pool))
